spotbugs/src/main/java/edu/umd/cs/findbugs/ba/ValueRangeMap.java

```diff
@@ -0,0 +1,86 @@
+package edu.umd.cs.findbugs.ba;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.ba.vna.ValueNumber;
+
+public class ValueRangeMap {
+    private Map<ValueNumber, LongRangeSet> variableRanges;
+    private Branch branch;
+
+    public ValueRangeMap() {
+        variableRanges = new HashMap<>();
+    }
+
+    public ValueRangeMap(ValueRangeMap other) {
+        variableRanges = new HashMap<>(other.variableRanges);
+        branch = other.branch;
+    }
+
+    public LongRangeSet getRange(ValueNumber vna) {
+        return variableRanges.get(vna);
+    }
+
+    public void setRange(ValueNumber vna, LongRangeSet range) {
+        variableRanges.put(vna, range);
+    }
+
+    public Branch getBranch() {
+        return branch;
+    }
+
+    public void setBranch(@Nullable Branch branch) {
+        this.branch = branch;
+    }
+
+    public void copyFrom(ValueRangeMap source) {
+        variableRanges = new HashMap<ValueNumber, LongRangeSet>(source.variableRanges);
+        branch = source.branch;
+    }
+
+    public void clear() {
+        variableRanges.clear();
+        branch = null;
+    }
+
+    public void meetWith(ValueRangeMap other) {
```

Comment:
I saw `meetWith()` in `LockSet` which is the _fact_ for `LockAnalysis`. Thus since `ValueRangeMap` is the _fact_ for `ValueRangeAnalysis` I used the same name. However, I can change it if you think `mergeWith` is a better name.